src/SegmentedMsg.chpl

```diff
@@ -1178,6 +1178,19 @@ module SegmentedMsg {
       smLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
       return new MsgTuple(repMsg, MsgType.NORMAL);
   }
+
+  proc stringsToJSONMsg(cmd: string, name: string, st: borrowed SymTab): MsgTuple throws {
+    var strings = getSegString(name, st);
+    var size = strings.size;
+    var rtn: [0..#size] string;
+
+    for i in 0..#size {
+      rtn[i] = strings[i];
+    }
```

Comment:
is there a reason not to do a `forall` here?

Reply:
Order matters. As far as I could find, there's no way to guarantee the order of the list matches the occurrence of the strings in the Strings object when using a forall loop.

Reply:
I'm pretty sure that shouldn't matter here because we're explicitly specifying the position in both arrays and every iteration is independent of the others and doesn't rely on the result from those iterations.

If we had something like:
```chapel
forall i in 0..#size {
  rtn += strings[i];
}
```
then order would certainly matter because then the `rtn` array depends on the order we visit the indices in the range `0..#size` (i.e. if the first iteration of our parallel loop is `i=27`, then the first index of `rtn` will be `strings[27]`

but since we are specifying `rtn[i]` this won't matter. Because if the first iteration of parallel loop is `i=27` then `rtn[27] = strings[27]`, so every index still has the right thing written into it. We don't need to know the result of any other loop iterations, so the order that we iterate through the loop shouldn't matter

Reply:
You're right, updated to use `forall`